src/spin_lock.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (c) 2014-2015 DataStax
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifndef __CASS_SPINLOCK_HPP_INCLUDED__
+#define __CASS_SPINLOCK_HPP_INCLUDED__
+
+#include <boost/atomic.hpp>
+
+namespace cass {
+
+class Spinlock {
+public:
+  typedef enum { LOCKED, UNLOCKED } LockState;
+
+  Spinlock()
+    : state_(UNLOCKED) {}
+
+  void lock() {
+    while (state_.exchange(LOCKED, boost::memory_order_acquire) == LOCKED) { }
+  }
+
+  void unlock() {
+    state_.store(UNLOCKED, boost::memory_order_release);
+  }
+
+private:
+  boost::atomic<LockState> state_;
+
+public:
+  typedef char CachePad[64];
+  CachePad pad__;
```

Comment:
+1 That is an interesting read. Thanks. Here's the program I was using to test contention/performance. I've modified it to toggle the cache padding: https://gist.github.com/mpenick/327ab00f4358d275cf66

It looks like a wash either way. I can see any noticeable difference, but my measurement of the average is kind of  ghetto. This looks like a job for HDR Histogram. :)